content/en/docs/usage/make_build.md

```diff
@@ -0,0 +1,352 @@
+In this example, we add the `lightweight IP` (`lwip`) network stack [`lib-lwip`](https://github.com/unikraft/lib-lwip):
+
+```console
+$ cd my-unikernel
+$ cd libs
+$ git clone https://github.com/unikraft/lib-lwip
+```
+
+Next, we have reference the library in the `Makefile` of `app-helloworld` so that it becomes available for selection in the configuration:
+
+```Makefile
+UK_LIBS ?= $(PWD)/../../libs
+
+LIBS := $(UK_LIBS)/lib-lwip
+#LIBS := $(LIBS-y):$(UK_LIBS)/lib-X
+```
+
+We can now re-run
+
+```console
+$ make menuconfig
+```
+
+and select `lwip` in the `Library Configuration` sub-menu.
+We can then rebuild the unikernel, by running:
+
+```console
+$ make clean
+$ make prepare
+$ make
+```
+
+The `prepare` step gives the build system the chance to pull the actual source code for `lwip` and apply necessary patches to make it work with Unikraft.
+This has to be done only once after adding a new library.
+
+Note that the build may fail because depending on the configuration, `lwip` requires a more extensive `libc` (e.g. [`musl`](https://github.com/unikraft/lib-musl)).
+You can add these additional dependencies just as described.
+
+To add new source files to the build system, we add them to the `Makefile.uk` file as such:
+
+```makefile
+$(eval $(call addlib,apphelloworld))
+
+APPHELLOWORLD_SRCS-y += $(APPHELLOWORLD_BASE)/main.c
+APPHELLOWORLD_SRCS-y += $(APPHELLOWORLD_BASE)/source-file-2.c
+```
+
+For more detailed information about using and configuring the build system, please refer to the [porting page](docs/develop/porting/).
+
+## Example: Using Make with app-helloworld-cpp
+
+Following the [steps above](docs/usage/make_build/#building-app-helloworld-with-make), we will setup, configure, build and run the [`app-helloworld-cpp` application](https://github.com/unikraft/app-helloworld-cpp).
+
+To create the setup, we will start with the following directory structure:
+
+```text
+my-unikernel/
+	|-- apps/
+	|   `-- app-helloworld-cpp      <- helloworld-cpp application
+	|-- libs/                       <- additional libraries go here
+	`-- unikraft/                   <- Unikraft core
+```
+
+We can do this by executing the following sequence of commands:
+
+```console
+$ mkdir my-unikernel
+$ cd my-unikernel
+$ git clone https://github.com/unikraft/unikraft
+$ mkdir libs
+$ mkdir apps
+$ cd apps
+$ git clone https://github.com/unikraft/app-helloworld-cpp
+$ cd app-helloworld-cpp
+```
+
+We can see that there is no `Makefile`, so we create one and add the external dependencies.
+
+```Makefile
+UK_ROOT ?= $(PWD)/../../unikraft
+UK_LIBS ?= $(PWD)/../../libs
+LIBS := $(UK_LIBS)/musl:$(UK_LIBS)/libunwind:$(UK_LIBS)/libcxx:$(UK_LIBS)/libcxxabi:$(UK_LIBS)/compiler-rt
+
+all:
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS)
+
+$(MAKECMDGOALS):
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS) $(MAKECMDGOALS)
+```
+
+After that, we configure and build the application:
+
+```console
+$ make menuconfig # Configure the application
```

Comment:
Split the configuration steps. Below `make menuconfig` list the options to select. Or, better yet, provide the `Config.uk` and then run `make menuconfig` and simply save.